### PR TITLE
[AOTI] Fix assert_function call in cpu autotune template

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #135086

Summary: In the ABI-compatible mode, assert_function should be AOTI_TORCH_CHECK.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @chauhang

### CODE REVIEW

File: test/inductor/test_cpu_cpp_wrapper.py [hunk 1]
```diff
@@ -103,6 +97,11 @@ class DynamicShapesCppWrapperCpuTests(InductorTestCase):
         ] = test_torchinductor.TestFailure(("cpp_wrapper",), is_skip=False)
     skip_list = [
         "test_multihead_attention_cpu",
+        *[
```
Comment (desertfire) [inline test/inductor/test_cpu_cpp_wrapper.py:100]: Move to the skip list and will be removed in the next PR

**Review by chenyang78 — APPROVED** [review]

**Review by angelayi — APPROVED** [review]

**Pull request closed without merging**